lib/Traits/TraitsClasses/CkanResourceFormatTraits.ts

```diff
@@ -14,11 +14,41 @@ export default class CkanResourceFormatTraits extends ModelTraits {
   @primitiveTrait({
     name: "Format Regular Expression",
     description:
-      "A regular expression that is matched against the distribution's format.",
+      "A regular expression that is matched against the distribution's format. This must be defined for this format to be used",
     type: "string"
   })
   formatRegex?: string;
 
+  @primitiveTrait({
+    name: "URL Regular Expression",
+    description:
+      "A regular expression that is matched against the url, this will only be used if `formatRegex` matches.",
+    type: "string"
+  })
+  urlRegex?: string;
+
+  @primitiveTrait({
+    name: "Maximum file size (MB)",
+    description:
+      "The maximum allowed file size for this format (in megabytes).",
+    type: "number"
+  })
+  maxFileSize?: number;
```

Comment:
Should this be nullable for it to be able to override the defaults?

Reply:
Yeah good idea

Reply:
done!